pkg/controller/datasource-controller.go

```diff
@@ -215,7 +246,35 @@ func addDataSourceControllerWatches(mgr manager.Manager, c controller.Controller
 		predicate.Funcs{
 			CreateFunc: func(e event.CreateEvent) bool { return true },
 			DeleteFunc: func(e event.DeleteEvent) bool { return true },
-			UpdateFunc: func(e event.UpdateEvent) bool { return false },
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				pvcOld, okOld := e.ObjectOld.(*corev1.PersistentVolumeClaim)
+				pvcNew, okNew := e.ObjectNew.(*corev1.PersistentVolumeClaim)
+				return okOld && okNew && pvcOld.Status.Phase != pvcNew.Status.Phase
+			},
+		},
+	); err != nil {
+		return err
+	}
+
+	if err := mgr.GetClient().List(context.TODO(), &snapshotv1.VolumeSnapshotList{}); err != nil {
+		if meta.IsNoMatchError(err) {
+			// Back out if there's no point to attempt watch
```

Comment:
Why should `NoMatch` be quitely ignored? for storage with no VolumeSnapshot supported?

Reply:
some clusters simply don't have the snapshot crds (for example the hpp csi lane)

Reply:
ack